compiler/rustc_ast/src/ast.rs

```diff
@@ -1518,6 +1518,33 @@ pub enum Movability {
     Movable,
 }
 
+/// Closure lifetime binder, `for<'a, 'b>` in `for<'a, 'b> |_: &'a (), _: &'b ()|`.
+#[derive(Clone, Encodable, Decodable, Debug)]
+pub enum ClosureBinder {
+    /// The binder is not present, all closure lifetimes are inferred.
+    NotPresent,
+    /// The binder is present.
+    For {
+        id: NodeId,
```

Comment:
Where is the `NodeId` used?

Reply:
To get the `HirId` later

Reply:
Removed in b30f7b27effda91a018a0222643ba192c5b6a8fe